### PR TITLE
preserve log selections; filter enhancements

### PR DESCRIPTION
**Pull request by pq**
preserve log selections / filter enhancements

* shows log counts for “empty” filters
* preserves selections on tree changes
* removes defunct entry filter class
* optimizes empty filter case
* mitigates occasional range check failure on tree initialization

/cc @jacob314 @devoncarew

### CODE REVIEW

File: src/io/flutter/logging/FlutterLogTree.java [hunk 1]
```diff
@@ -411,6 +413,9 @@ void update() {
 
       reload(getRoot());
       treeTable.updateUI();
+      // Preserve / restore selection state.
+      treeTable.getTree().setSelectionRows(treeTable.getSelectedRows());
```
Comment (jacob314) [inline src/io/flutter/logging/FlutterLogTree.java]: I'm a little surprised this is line is required.

Comment (pq) [reply src/io/flutter/logging/FlutterLogTree.java]: Me too.   There's something about the UI update that does not preserve the selection.  I tried a bunch of different things and this is the only thing that worked.

File: src/io/flutter/logging/FlutterLogTree.java [hunk 2]
```diff
@@ -706,31 +675,33 @@ public void setFilter(@NotNull FlutterLogFilterPanel.FilterParam filter) {
...
+      try {
+        return RowFilter.regexFilter(
+          standardRegex,
+          FlutterTreeTableModel.ColumnIndex.MESSAGE.index,
+          FlutterTreeTableModel.ColumnIndex.CATEGORY.index,
+          FlutterTreeTableModel.ColumnIndex.LOG_LEVEL.index
+        );
+      }
+      catch (PatternSyntaxException e) {
+        // Fall through.
```
Comment (jacob314) [inline src/io/flutter/logging/FlutterLogTree.java]: not a new issue but we should notify the user if the regex is invalid.

File: src/io/flutter/logging/FlutterLogTree.java [hunk 3]
```diff
@@ -747,6 +718,8 @@ private void updateCounter() {
   }
 
   public void clearEntries() {
+    // Ensure selections are cleared on empty.
+    getTree().setSelectionRows(new int[]{});
```
Comment (jacob314) [inline src/io/flutter/logging/FlutterLogTree.java]: use 
    this.getTree().clearSelection();

instead

Comment (pq) [reply src/io/flutter/logging/FlutterLogTree.java]: Good idea.

File: src/io/flutter/logging/FlutterLogTree.java (hunk 2 above)
Comment (pq) [reply src/io/flutter/logging/FlutterLogTree.java]: Yeah.  I'm not sure what the best UI is but ignoring isn't great.  I'll add a TODO.

**Pull request merged**